### DISCUSSION TITLE
How to generate custom class in output css?

### DISCUSSION THREAD

**Discussion by bldrdash** [unocss/unocss#3276] [Q&A]
### UnoCSS version
0.57.1

I'm attempting to switch from tailwindcss to unocss, but am having a problem with generating the css output file as expected. What I'm seeing is tailwindcss will generate an output css file that contains the custom class name ```fooclass```, and the transformed @apply.

When using unocss, I see it picks up the contents of the apply (```.bg-blue-500```), but doesn't output the ```fooclass```.

### Reproduction 

my.css:
```css
.fooclass {
  @apply bg-blue-500;
}
```

With tailwind:
```bash
tailwindcss init
tailwindcss -o out.css -i my.css
```
The resulting out.css is:
```css
.fooclass {
  --tw-bg-opacity: 1;
  background-color: rgb(59 130 246 / var(--tw-bg-opacity))
}
```
In uno using the same my.css:

```ts
// uno.cnfig.ts:
import { defineConfig, presetUno } from 'unocss'

export default defineConfig({
  content: {
    filesystem: [
      '*.css',
    ]
  },
  presets: [
    presetUno(),
  ],
})
```
```bash
unocss -o out.css my.css
tail -n1 out.css
> .bg-blue-500{--un-bg-opacity:1;background-color:rgb(59 130 246 / var(--un-bg-opacity));}
grep fooclass out.css || echo "Not found"
> Not found
```
Is there a way to get unocss to include my ```fooclass` in the ```out.css``` file?

### System info
Debian 11 bash
No build framework (just using command line utilities)
Tailwindcss v3.3.4

**Comment by chu121su12** [comment]
I'm not used to use cli, but I think you can workaround this with the `selector-x` variant, so to write a very verbose:
```scss
.fooclass {
  @apply selector-[.fooclass]-bg-blue-500;
}
```
Use [cautiously](https://unocss.dev/play/?html=DwEwlgbgBAxgNgQwM5ILwCIkFM5ZgFwHsAnAWgG0A6AM0MPmSQF1SAjAczbgFctSBWAAyD0APiA&config=PTAEFMGcBsEsDsAuBaAJrSBDARtcz5wAPFOQ0eAe2QCcpEbYBjRcVZWAWwAdKbFIAKC69%2BoAN6DQoVOABmCcAGFK8BQHMANFNDc6kcIgCCiBrGwBXRLDkBPbdL1RDASSarID3fsMBVKtoAvqByNJScoADkFlRMkJCRgoLEoogy8pgW0GmyCoQqarDqABSS0jRZUABcoADaOtK1kUwWkIjhtJWRmhKg7tB8NZF0qJGggQC6XlM6kAAWfIgtAjVl0lEtbR3zi8uRQ6wkyNDqoIcofJjw6uCgCwBu4DRV58ismNDdOoFeTgYrdQa3mciH8lGKAEovI4fMZTIxLNY7JDocD-m4PKUgdJIEwPuAagBGAB0ACZUdImKh4EM5qZuJAqiAoJxifNgF91uMoToZoEIYIgA&css=PQKgBA6gTglgLgUzAYwK4Gc4HsC2YDCAyoWABYJQIA0YAhgHYAmYcUD6AZllDhWOqgAOg7nAB0YAGLcwCAB60cggDYIAXGBDAAUKDBi0mXGADe2sGC704AWgDuCGAHNScDQFYADJ4Dc5sAACtMLKAJ5gggCMLPK2ABR2pPBIcsoAlH4WAEa0yADWTlBYqEw2yFjK3Bpw5LxxAOTllVDoYpSMYgAs3vUZ2gC%2BmsBAA&options=N4IgLgTghgdgzgMwPYQLYAkyoDYgFwJTZwCmAvkA).

**Reply by bldrdash** [reply]
Thanks for your response.   I found that using that syntax in the .css file didn't result in the fooclass being generated in the output, but simply using this syntax in the html:
```css
<div class="selector-[.fooclass]-bg-blue-500">Managed by.fooclass</div>
```
did.

However, I was looking for better compatibility with tailwinds in the hopes of using unocss with minimal conversion, and the proposed solution is, as you mentioned, pretty verbose.
